WordPress/gutenberg
# Query: Ensure create new post link is above block styles

Fixes: https://github.com/WordPress/gutenberg/issues/49327

## What?

Moves the query block's "Create new post" link so that it will render above any block styles registered by plugins.

## Why?

The block inspector's layout appears broken when the "Create new post" link doesn't immediately follow the block card and any block variation transforms.

## How?
- Moves the `CreateNewPostLink` component from the query block into the block editor package
- Updates `CreateNewPostLink` to retrieve the attributes from the stores currently selected block
- Renders the new `CreateNewPostLink` component in the block inspector for single block selections when the current block is a Query block
- Removes the old component, hook file, and filter that previously injected the create new post link.

#### Alternate Approach

One possibility I looked into was to resurrect a previously tried [SlotFill approach](https://github.com/WordPress/gutenberg/pull/45437/) but this time lock it as a private API. This turned out to be a much heavier change, touching more areas, and still wouldn't 100% prevent the concerns that blocked #45437.

## Testing Instructions
1. Add some block styles to the Query block (example in snippet below)
2. Edit a post, add a query block, select it and confirm the incorrect create new post link position
3. Checkout this PR branch, rebuild, and reload the post
4. The create new post link position should now be below the block card but above the block styles

```php
<?php
/**
 * Register block styles.
 */
function blue_note_register_block_styles() {
	register_block_style(
		'core/query',
		array(
			'name'  => 'blue-note-query-slant',
			'label' => __( 'Slanted images', 'blue-note' ),
			'inline_style' => '.is-style-blue-note-query-slant .wp-block-post-featured-image {transform: rotate(-1deg);}',
		)
	);
}
add_action( 'init', 'blue_note_register_block_styles' );
```

## Screenshots or screencast 

| Before | After |
|---|---|
| <img width="280" alt="Screenshot 2023-04-04 at 3 53 24 pm" src="https://user-images.githubusercontent.com/60436221/229701522-46eaae64-29a6-4651-b0d7-3bce388ed545.png"> | <img width="282" alt="Screenshot 2023-04-04 at 3 52 29 pm" src="https://user-images.githubusercontent.com/60436221/229701546-9e9ce8c8-9b86-449a-a39e-674731a8b8b6.png"> |
